core/eds_test.go

```diff
@@ -28,16 +27,19 @@ func TestTrulyEmptySquare(t *testing.T) {
 }
 
 // TestNonEmptySquareWithZeroTxs tests that a non-empty square with no
-// transactions or blobs computes the correct data root (not the minimum DAH).
+// transactions or blobs computes the correct data root the minimum DAH.
+// Technically, this block data is invalid because the construction of the
+// square is deterministic, and the rules which dictate the square size do not
+// allow for empty block data. However, should that ever occur, we need to to
+// ensure that the correct data root is generated.
 func TestNonEmptySquareWithZeroTxs(t *testing.T) {
 	data := types.Data{
 		Txs:        []types.Tx{},
-		Blobs:      []types.Blob{},
 		SquareSize: 16,
 	}
 
 	eds, err := extendBlock(data)
 	require.NoError(t, err)
 	dah := da.NewDataAvailabilityHeader(eds)
-	assert.NotEqual(t, share.EmptyRoot().Hash(), dah.Hash())
+	assert.Equal(t, share.EmptyRoot().Hash(), dah.Hash())
```

Comment:
note that square construction is now deterministic! meaning that given some set of transactions, there only exists a single valid square. Therefore adding square size to the block data should still result in an empty DAH.